CoreFoundation/Base.subproj/CoreFoundation_Prefix.h

```diff
@@ -263,7 +265,15 @@ void OSMemoryBarrier();
 
 #endif
 
-#if DEPLOYMENT_TARGET_WINDOWS || DEPLOYMENT_TARGET_LINUX    
+#if DEPLOYMENT_TARGET_CYGWIN
+#define HAVE_STRUCT_TIMESPEC 1
+#define strncasecmp_l(a, b, c, d) strncasecmp(a, b, c)
+#define _NO_BOOL_TYPEDEF
+#include <windows.h>
```

Comment:
Functions ```eventfd(), epoll(), timerfd_create()``` used for Linux are not supported by Cygwin because they are not POSIX standard. I used the Windows implementation using ```#elif DEPLOYMENT_TARGET_WINDOWS || DEPLOYMENT_TARGET_CYGWIN```.
Win32 API is used only for ```CFRunLoop.c```.